### PR TITLE
Update Index.rst (about JSON response)

### PR DESCRIPTION
**Pull request by hmccloy**
There is an error in the doc:
For a JSON respone your have to use JsonResponse (which implements Response)

### CODE REVIEW

**Comment by mbrodala** [comment]
Actually you should not use the `JsonResponse` class since it is `@internal`:

https://github.com/TYPO3/TYPO3.CMS/blob/10.4/typo3/sysext/core/Classes/Http/JsonResponse.php#L23

Instead you should use the `ResponseFactoryInterface` as described here:

https://docs.typo3.org/m/typo3/reference-coreapi/10.4/en-us/ApiOverview/RequestHandling/Index.html#creating-new-request-response-objects

Can you adjust the example and link to that page instead?

File: Documentation/ApiOverview/Ajax/Backend/Index.rst [hunk 1]
```diff
@@ -103,7 +104,7 @@ $reasonPhrase
    {
        // our previous computation
 
-       return new Response(json_encode(['result' => $result]), 200, ['Content-Type' => 'application/json; charset=utf-8']);
+       return new JsonResponse(['result' => $result], 200);
```
Comment (mbrodala) [inline Documentation/ApiOverview/Ajax/Backend/Index.rst:107]: See comment.

**Review by mbrodala — CHANGES_REQUESTED** [review]

**Comment by marble** [comment]
@hmccloy 

> Can you adjust the example and link to that page instead?

**Comment by linawolf** [comment]
I had to integrate the commit in another PR, so that I can edit it. See #1223

**Pull request closed without merging**